File: RayTracer/Material.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace RayTracer
{
    public class Material
    {

        public Material()
        {
            Specular = new MyColor();
            Diffuse = new MyColor();
            Emission = new MyColor();
            Shininess = 0f;
        }

        public MyColor Specular
        { get; set; }

        public MyColor Diffuse
        { get; set; }

        public MyColor Emission
        { get; set; }

        public float Shininess
        { get; set; }

        public Material Clone()
        {
            return (Material)this.MemberwiseClone();
        }
    }
}

File: RayTracer - CPP - BVH - OBJ/RayTracer/Material.cpp
#include "Material.h"



Material::Material()
{
	specular = MyColor();
	diffuse = MyColor();
	emmission = MyColor();
	ambient = MyColor();

}

void Material::setShininess(float val)
{
	shininess = val;
	if (shininess < 0)
		shininess = 0;
	if (shininess > 255)
		shininess = 255;
}

void Material::setrefIndex(float val)
{
	refractIndex = val;
}

void Material::setRefValue(float val)
{
	refrVal = val;
	if (refrVal < 0)
		refrVal = 0;
	if (refrVal > 1)
		refrVal = 1;
}


Material::~Material()
{
}
